File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func rightSideView(root *TreeNode) []int {
    if root == nil {
        return []int{}
    }
    
    arr := []*TreeNode{root}
    ans := []int{}
    for len(arr) > 0 {
        ans = append(ans, arr[len(arr)-1].Val)
        newArr := make([]*TreeNode, 0)
        for _, v := range arr {
            if v.Left != nil {
                newArr = append(newArr, v.Left)
            }

            if v.Right != nil {
                newArr = append(newArr, v.Right)
            }
        }

        arr = newArr
    }

    return ans
}

File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        if not root:
            return []

        ans = []
        q = collections.deque([root])

        while q:
            size = len(q)
            for i in range(size):
                root = q.popleft()
                if i == size - 1:
                    ans.append(root.val)
                if root.left:
                    q.append(root.left)
                if root.right:
                    q.append(root.right)

        return ans
        